src/main/java/co/rsk/federate/FederationProviderFromFederatorSupport.java

```diff
@@ -135,9 +143,58 @@ public Optional<Address> getRetiringFederationAddress() {
         return federatorSupport.getRetiringFederationAddress();
     }
 
+    @Override
+    public Optional<Federation> getProposedFederation() {
+        if (!federatorSupport.getConfigForBestBlock().isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+       
+        Integer federationSize = federatorSupport.getProposedFederationSize()
+            .orElse(FEDERATION_NON_EXISTENT.getCode());
+        if (federationSize == FEDERATION_NON_EXISTENT.getCode()) {
+            return Optional.empty();
+        }
```

Comment:
This would work if the Bridge returned only a null value for the federation size if the proposed fed did not exist. But it can return FEDERATION_NON_EXISTENT which would be present in the optional. We would have to check this before continuing.

Reply:
pretty sure the `FEDERATION_NON_EXISTENT` code is not being used for the proposed fed, but for the pending and retiring.

Reply:
It was implemented [here](https://github.com/rsksmart/rskj/pull/2794/files#diff-34109dfc71ff55ae26e9ede5ada468333341d0be091e537ad21ccd847c57351fR1082).

Reply:
my bad, you right. But in that case it wont be necessary to have an optional right? Because you are always getting a value from the Bridge

Reply:
The Bridge is always returning something, but the result of evaluating the callTx can be null for variety of reasons. With the optional you avoid this error handling and then when it is time to consume it you explicitly declare how you want to interpret a missing value. In this case we want to reinforce that it should be FEDERATION_NON_EXISTENT code.

Reply:
I get it. But i don't like that we are casting a null response (that will come from an error, since we are returning `-1` when the proposed fed does not exist) to `FEDERATION_NON_EXISTENT`. I find it confusing, makes me think that the Bridge will return an empty response when there is no proposed fed.